setup.py

```diff
@@ -33,52 +58,13 @@ def is_numpy_installed():
     author_email='braxton.mckee@gmail.com',
     url='https://github.com/braxtonmckee/nativepython',
     packages=setuptools.find_packages(),
-    ext_modules=[
-        setuptools.Extension(
-            'typed_python._types',
-            sources=[
-                'typed_python/all.cpp',
-            ],
-            define_macros=[
-                ("_FORTIFY_SOURCE", 2)
-            ],
-            extra_compile_args=[
-                '-O2',
-                '-fstack-protector-strong',
-                '-Wformat',
-                '-Wdate-time',
-                '-Werror=format-security',
-                '-std=c++14',
-                '-Wno-sign-compare',
-                '-Wno-narrowing',
-                '-Wno-unused-variable'
-            ],
-            # pkg_resources.resouce_filename raises an exception if numpy is
-            # not installed, so we check if it is installed. This works because
-            # setup.py is called first to collect and install dependencies,
-            # and then to install the package itself.
-            include_dirs=[
-                pkg_resources.resource_filename('numpy', 'core/include')
-            ] if is_numpy_installed() else []
-        )
-    ],
-    install_requires=[
-        'boto3',
-        'flask-login',
-        'flask-sockets',
-        'flask-wtf',
-        'gevent',
-        'ldap3',
-        'llvmlite',
-        'lz4',
+    cmdclass={'build_ext': NumpyBuildExtension},
+    ext_modules=ext_modules,
+    setup_requires=[
         'numpy',
-        'psutil',
-        'pytz',
-        'pyyaml',
-        'redis',
-        'requests',
-        'websockets',
+        'flask'
```

Comment:
I know why setup requires numpy, but why does it require flask?

Reply:
flask and friends are required in order to properly run the tests (object database and web tools use it). We can remove them from this setup file, I think, but we'd have to keep them in the requirements/pipfile